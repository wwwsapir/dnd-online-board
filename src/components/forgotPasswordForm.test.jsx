import ForgotPasswordForm from "./ForgotPasswordForm";
import { MemoryRouter } from "react-router-dom";
// Note: I use this specific test file as a test template for other tests creation, so there are lot of redundant comments here.

// Shallow rendering test
it("renders correctly - shallow", () => {
  const wrapper = shallow(
    <ForgotPasswordForm
    // prop={prop hard coded value}
    // function_prop={() => empty/test function}
    />
  );

  expect(wrapper).toMatchSnapshot();
});

// Test send click functionality (async)
jest.mock("../services/sendResetPasswordEmail");
it("handles Send click correctly", async (done) => {
  const wrapper = mount(
    <MemoryRouter>
      <ForgotPasswordForm />
    </MemoryRouter>
  );
  expect(wrapper.find("label").first().text()).toEqual(
    "A message with a link to reset your password will be sent to this email address."
  );
  wrapper
    .find("input")
    .first()
    .simulate("change", {
      target: { name: "email", value: "wwwsapir@gmail.com" },
    });
  wrapper.find("button").simulate("submit");
  setTimeout(() => {
    wrapper.update();
    expect(wrapper.find("label").first().text()).toEqual(
      "Reset message sent to email address!"
    );
    done();
  });
});

// Deep html inner elements rendering test
// it("renders correctly - render", () => {
//   const wrapper = render(<ForgotPasswordFor />);

//   expect(wrapper).toMatchSnapshot();
// });

// Specific inner elements value test
// it("renders correctly - mount", () => {
//   const wrapper = mount(<ForgotPasswordForm />);

//   // *here we can grab a specific DOM element and check its value

//   expect(wrapper).toMatchSnapshot();
// });

// Trigget some clicks and check the values in elements or that a prop function was called
// it("calls prop function on click", () => {
//   const spy = sinon.spy();
//   const wrapper = mount(
//     <ForgotPasswordForm
//     functionProp={spy}
//     />
//   );

//   wrapper.find("div").first().simulate("click");

//   expect(spy.calledOnce).toBe(true);
// });
