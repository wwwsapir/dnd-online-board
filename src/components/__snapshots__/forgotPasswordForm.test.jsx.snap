// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - shallow 1`] = `
<div
  className="menu-window"
>
  <ul
    className="menu bg-dark w-100"
  >
    <h4
      className="mb-5"
    >
      <span
        className="menu-header"
      >
        Reset Password
      </span>
    </h4>
    <li>
      <input
        className="input-group-sm form-control"
        id="email"
        onChange={[Function]}
        placeholder="Email Address"
        required={true}
        value=""
      />
    </li>
    <li>
      <label>
        A message with a link to reset your password will be sent to this email address.
      </label>
    </li>
    <li>
      <button
        className="btn btn-primary form-control mt-3"
        disabled={false}
        onClick={[Function]}
      >
        Send
      </button>
    </li>
    <li
      className="mt-2 mb-4"
    >
      <Link
        className="float-right"
        to="/home/login"
      >
        Back to log in
      </Link>
    </li>
  </ul>
</div>
`;
